libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/floatingvideo/FloatingVideoService.kt

```diff
@@ -0,0 +1,727 @@
+                            toggleFullScreen()
+                        }
+                        true
+                    }
+
+                    else -> false
+                }
+            }
+        }
+        overlayContainer?.addView(maximizeButton)
+
+        val controlBar = LinearLayout(this).apply {
+            orientation = LinearLayout.HORIZONTAL
+            gravity = Gravity.CENTER_VERTICAL
+            layoutParams = FrameLayout.LayoutParams(
+                FrameLayout.LayoutParams.MATCH_PARENT,
+                dpToPx(48) // height of bar
+            ).apply {
+                gravity = Gravity.BOTTOM
+            }
+            setPadding(dpToPx(8), dpToPx(4), dpToPx(8), dpToPx(4))
+            setBackgroundColor(Color.parseColor("#80000000")) // optional semi-transparent bg
+        }
+        playPauseButton = ImageView(this).apply {
+            setImageResource(android.R.drawable.ic_media_pause)
+            setColorFilter(Color.WHITE)
+            background = GradientDrawable().apply {
+                shape = GradientDrawable.OVAL
+                setColor(Color.parseColor("#40FFFFFF")) // semi-transparent background
+            }
+            layoutParams = LinearLayout.LayoutParams(dpToPx(20), dpToPx(20)).apply {
+
+            }
+
+            setOnClickListener {
+                videoView?.let { vv ->
+                    if (vv.isPlaying) {
+                        vv.pause()
+                        setImageResource(android.R.drawable.ic_media_play)
+                    } else {
+                        setImageResource(android.R.drawable.ic_media_pause)
+                        vv.start()
+
+                    }
+                }
+            }
+        }
+
+        seekBar = SeekBar(this).apply {
+            visibility = View.VISIBLE
+            layoutParams = LinearLayout.LayoutParams(
+                0, dpToPx(32), 1f // take remaining width
+            ).apply {
+                setMargins(8,0,8,0)
+            }
+            setPadding(dpToPx(16), dpToPx(0), dpToPx(16), dpToPx(0))
+
+        }
+
+        controlBar.addView(playPauseButton)
+        controlBar.addView(seekBar)
+
+// Add control bar to container
+        container.addView(controlBar)
+        seekBar?.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                if (fromUser) {
+                    videoView?.seekTo(progress)
+                }
+            }
+
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {
+                // pause updates while user is dragging
+                progressHandler.removeCallbacksAndMessages(null)
+            }
+
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {
+                startProgressUpdater()
+                showControls()
+            }
+        })
+
+        container.addView(overlayContainer)
+        return container
```

Comment:
It would be awesome to use Compose UI to design the View. It will simplify the code a lot (I think), and allow to have previews. Maybe https://gist.github.com/kishan-vadoliya/9fbd1e3c1590de1e4a1a830c5d4edb3f can inspire?